### PR TITLE
Move charts to templates

### PR DESCRIPTION
**Pull request by lucchmielowski**
# Description

Attempts at moving to monolithic chart from subcharts to make our life easier for now

### CODE REVIEW

**Review by eddycharly — APPROVED** [review]

**Review by Atrax1 — CHANGES_REQUESTED** [review]
Why not creating folder charts in nsq that contains nsqd, nsqlookupd and nsqadmin ?
It is not the way to put everything in templates.. Maybe I'm missing some reasons

**Comment by lucchmielowski** [comment]
@Atrax1 Well, we tried putting everything in `/charts` but decided to move from that to a monolith for now as it's a real pain in the ass to maintain, for example when working with local `dependencies` and `_helpers` files. If we wanted to have a nsq _umbrella chart_ the real way to do it would be to have something like : 
```
├── charts
│   ├── nsq
│   │   ├── Chart.yaml # umbrella chart requiring the 3 other charts
│   │   └── values.yaml
│   ├── nsqd
│   │   ├── Chart.yaml
│   │   └── values.yaml
│   ├── nsqadmin
│   │   ├── Chart.yaml
│   │   └── values.yaml
│   ├── nsqlookupd
│   │   ├── Chart.yaml
│   │   └── values.yaml
```

File: charts/nsq/templates/nsqadmin/tests/test-connection.yaml [hunk 1]
```diff
@@ -11,5 +11,9 @@ spec:
     - name: wget
       image: busybox
       command: ['wget']
+<<<<<<< HEAD:charts/nsq/charts/nsqadmin/templates/tests/test-connection.yaml
```
Comment (RiceBowlJr) [inline charts/nsq/templates/nsqadmin/tests/test-connection.yaml:14]: Oupsy

**Comment by Atrax1** [comment]
> @Atrax1 Well, we tried putting everything in `/charts` but decided to move from that to a monolith for now as it's a real pain in the ass to maintain, for example when working with local `dependencies` and `_helpers` files. If we wanted to have a nsq _umbrella chart_ the real way to do it would be to have something like :
> 
> ```
> ├── charts
> │   ├── nsq
> │   │   ├── Chart.yaml # umbrella chart requiring the 3 other charts
> │   │   └── values.yaml
> │   ├── nsqd
> │   │   ├── Chart.yaml
> │   │   └── values.yaml
> │   ├── nsqadmin
> │   │   ├── Chart.yaml
> │   │   └── values.yaml
> │   ├── nsqlookupd
> │   │   ├── Chart.yaml
> │   │   └── values.yaml
> ```
> 
> Since we do not want to separate those 3 entities ATM, we chose to put everything under the same chart for now and move it later if we need.

I was not seing it like that either you don't require in nsq the 3 other chart but it would rather be nsq is parent chart containing charts folder with those 3 charts

File: charts/nsq/templates/nsqadmin/tests/test-connection.yaml (hunk 1 above)
Comment (lucchmielowski) [reply charts/nsq/templates/nsqadmin/tests/test-connection.yaml:14]: Oh common git !

**Comment by lucchmielowski** [comment]
@Atrax1 From what I understand, what you mean is something that looks like : 

```
├── charts
│   ├── nsq
│   │   ├── charts
│   │   │   ├── nsqd
│   │   │   ├── nsqadmin
│   │   │   ├── nsqlookupd
│   │   └── values.yaml
```

but if you look at the PR, that's what we're trying to move from as using subchart like that forces us to create hack on the charts configuration (for example putting empty `repository` in dependencies, etc) to make stuff work (although the Helm doc says it's possible)

**Comment by Atrax1** [comment]
> @Atrax1 From what I understand, what you mean is something that looks like :
> 
> ```
> ├── charts
> │   ├── nsq
> │   │   ├── charts
> │   │   │   ├── nsqd
> │   │   │   ├── nsqadmin
> │   │   │   ├── nsqlookupd
> │   │   └── values.yaml
> ```
> 
> but if you look at the PR, that's what we're trying to move from as using subchart like that forces us to create hack on the charts configuration (for example putting empty `repository` in dependencies, etc) to make stuff work (although the Helm doc says it's possible)

Yes like this but I don't get it why we are talking about dependencies and hacks here. It is the proper way to do it

File: charts/nsq/templates/nsqadmin/tests/test-connection.yaml (hunk 1 above)
Comment (RiceBowlJr) [reply charts/nsq/templates/nsqadmin/tests/test-connection.yaml]: J'ai fix.

**Review by RiceBowlJr — APPROVED** [review]
OK for me. Agree on the principle (did struggle with you and did not find a chart done with subcharts in charts in any upstream charts 🤔).

**Pull request closed without merging**